src/Item/Item.service.ts

```diff
@@ -85,14 +94,211 @@ export class ItemService {
+  }
+
+  public async getTPItemsByManager(
+    manager: string
+  ): Promise<{
+    dbTPItems: ItemAttributes[]
+    remoteTPItems: ThirdPartyItemFragment[]
+  }> {
+    const thirdParties = await thirdPartyAPI.fetchThirdPartiesByManager(manager)
+    if (thirdParties.length <= 0) {
+      return { dbTPItems: [], remoteTPItems: [] }
+    }
+
+    const thirdPartyIds = thirdParties.map((thirdParty) => thirdParty.id)
+
+    const [dbTPItems, remoteTPItems] = await Promise.all([
+      Item.findByThirdPartyIds(thirdPartyIds),
+      thirdPartyAPI.fetchItemsByThirdParties(thirdPartyIds),
+    ])
+
+    return { dbTPItems, remoteTPItems }
+  }
+
+  public splitItems(
+    allItems: ItemAttributes[]
+  ): { items: ItemAttributes[]; tpItems: ItemAttributes[] } {
+    const items: ItemAttributes[] = []
+    const tpItems: ItemAttributes[] = []
+
+    for (const item of allItems) {
+      if (isTPItem(item)) {
+        tpItems.push(item)
+      } else {
+        items.push(item)
+      }
+    }
+
+    return { items, tpItems }
+  }
+
+  public withOwner<T extends FullItem | ItemAttributes>(
+    items: T[],
+    ethAddress: string
+  ) {
+    return items.map((item) => ({ ...item, eth_address: ethAddress }))
+  }
+
+  private async getTPCollectionItems(
+    dbCollection: ThirdPartyCollectionAttributes,
+    dbItems: ItemAttributes[]
+  ): Promise<{ collection: CollectionAttributes; items: FullItem[] }> {
+    // TODO: This could be a single query, the problem is paginating the thing. We should only paginate remoteItems
+    const [lastItem, remoteItems] = await Promise.all([
+      thirdPartyAPI.fetchLastItem(
+        dbCollection.third_party_id,
+        dbCollection.urn_suffix
+      ),
+      thirdPartyAPI.fetchItemsByCollection(
+        dbCollection.third_party_id,
+        dbCollection.urn_suffix
+      ),
+    ])
+    const collection = lastItem
+      ? Bridge.mergeTPCollection(dbCollection, lastItem)
+      : dbCollection
+
+    const items = await Bridge.consolidateTPItems(dbItems, remoteItems)
+    return { collection, items }
+  }
+
+  private async getDCLCollectionItems(
+    dbCollection: CollectionAttributes,
+    dbItems: ItemAttributes[]
+  ): Promise<{ collection: CollectionAttributes; items: FullItem[] }> {
+    const {
+      collection: remoteCollection,
+      items: remoteItems,
+    } = await collectionAPI.fetchCollectionWithItemsByContractAddress(
+      dbCollection.contract_address!
+    )
+    const collection = remoteCollection
+      ? Bridge.mergeCollection(dbCollection, remoteCollection)
+      : dbCollection
+
+    const items = await Bridge.consolidateItems(dbItems, remoteItems)
+
+    return { collection, items }
+  }
+
+  private async getTPItem(
+    dbItem: ItemAttributes
+  ): Promise<{ item: FullItem; collection?: CollectionAttributes }> {
+    let item: FullItem = Bridge.toFullItem(dbItem)
+    let collection: CollectionAttributes | undefined
+
+    if (dbItem.collection_id) {
```

Comment:
Agree!, *but* considering this

![image](https://user-images.githubusercontent.com/692077/150022657-c536f540-c1d5-4e0c-8efa-58ca940c7fc6.png)

because of how the types are set up, we'd get that error (which is correct, `collection_id` could be `string | null`) So I'll change it to `ThirdPartyItemAttributes` which has the correct typing